src/sudoers/mod.rs

```diff
@@ -242,7 +307,11 @@ fn check_list_permission<User: UnixUser + PartialEq<User>>(
         .flatten()
         .fold(None::<Tag>, |outcome, (_, (tag, _))| {
             if let Some(outcome) = outcome {
-                let new_outcome = if !outcome.passwd { outcome } else { tag };
+                let new_outcome = if !outcome.passwd.unwrap_or(true) {
```

Comment:
I'd flip this conditional to avoid the negation given that you already modified it.